controllers/admin_domains.php

```diff
@@ -463,10 +468,33 @@ public function getModuleTlds()
                 $disabled = true;
             }
 
-            $tlds[$tld] = ['name' => $tld, 'disabled' => $disabled];
+            $response['tlds'][$tld] = ['name' => $tld, 'disabled' => $disabled];
+        }
+
+        // Build TLD terms
+        $response['terms'] = true;
+
+        // Check if the module supports pricing sync
+        try {
+            $module_rows = $this->ModuleManager->getRows($module->id);
+            $module_row = reset($module_rows) ?? (object) [];
+            $tld_prices = $this->ModuleManager->moduleRpc($module->id, 'getTldPricing');
```

Comment:
We're doing more processing here than we need.  Instead we can use a reflection class to see if the module implement the method.

```
            $class_name = Loader::toCamelCase($module->class);
            $reflection = new ReflectionClass($class_name);
            if ($reflection->getMethod('getFilteredTldPricing')->class !== $class_name) {
```